%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Wenneker Article
% LaTeX Template
% Version 2.0 (28/2/17)
%
% This template was downloaded from:
% http://www.LaTeXTemplates.com
%
% Authors:
% Vel (vel@LaTeXTemplates.com)
% Frits Wenneker
%
% License:
% CC BY-NC-SA 3.0 (http://creativecommons.org/licenses/by-nc-sa/3.0/)
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%----------------------------------------------------------------------------------------
%	PACKAGES AND OTHER DOCUMENT CONFIGURATIONS
%----------------------------------------------------------------------------------------

\documentclass[10pt, a4paper, twocolumn]{article} % 10pt font size (11 and 12 also possible), A4 paper (letterpaper for US letter) and two column layout (remove for one column)

\input{structure.tex} % Specifies the document structure and loads requires packages

%----------------------------------------------------------------------------------------
%	ARTICLE INFORMATION
%----------------------------------------------------------------------------------------

\title{Joint estimation of paternity, sibships and pollen dispersal in a snapdragon hybrid zone} % The article title

\author{
	\authorstyle{
        Thomas James Ellis\textsuperscript{1,2},
        David Luke Field, \textsuperscript{3,4},
        Nicholas H. Barton\textsuperscript{1,*}} % Authors
	\newline\newline % Space before institutions
	\textsuperscript{1}\institution{Institute of Science and Technology Austria, 2234 Klosterneuburg, Austria}\\ % Institution 1
	\textsuperscript{2}\institution{Gregor Mendel Institute of Molecular Plant Sciences, Doktor-Bohr-Gasse 3, 1030 Vienna, Austria}\\ % Institution 2
	\textsuperscript{3}\institution{Applied Biosciences, Macquarie University, Sydney, New South Wales, Australia}\\ % Institution 3
    \textsuperscript{4}\institution{School of Science, Edith Cowan University, Joondalup, Western Australia, Australia}\\ % Institution 4
    \textsuperscript{*}\institution{To whom correspondence should be addressed: nick.barton@ist.ac.at} 
}

% Example of a one line author/institution relationship
%\author{\newauthor{John Marston} \newinstitution{Universidad Nacional Autónoma de México, Mexico City, Mexico}}

\date{\today} % Add a date here if you would like one to appear underneath the title block, use \today for the current date, leave empty for no date

%----------------------------------------------------------------------------------------

\begin{document}

\maketitle % Print the title

\thispagestyle{firstpage} % Apply the page style for the first page (no headers and footers)
\linenumbers
%----------------------------------------------------------------------------------------
%	ABSTRACT
%----------------------------------------------------------------------------------------

% \lettrineabstract{}

\section{Abstract}
Inferring pedigrees of wild populations is useful because it gives direct estimates of mating patterns and variance in reproductive success.
Pedigree inference can be improved by including information about parentage shared between siblings, or by modelling phenotypes or population data related to mating.
However, we currently lack a framework to infer paternity, sibship, and population in a single analysis.
To address this we here extend a previous method Fractional Analysis of Paternity and Sibships to include population data.
We illustrate this with the example of pollen dispersal in a natural hybrid zone population of the snapdragon \textit{Antirrhinum majus}.
Pollen dispersal is leptokurtic, with half of mating events occurring within 30m, but with a long tail of mating events up to 759m.
Using simulations we find that both sibship and population information substantially improve pedigree reconstruction, and that we can expect to resolve median dispersal distances with high accuracy.

%----------------------------------------------------------------------------------------
%	ARTICLE CONTENTS
%----------------------------------------------------------------------------------------

\section{Introduction}

Knowledge of the genealogical relationships between individuals - their pedigree - is an extremely useful tool for understanding natural populations because it gives a direct estimate of who mates with whom, and differences in reproductive success \citep{thompson1976inference, pemberton2008wild}.
One example, pedigrees can tell us about the distribution of dispersal distances organisms or gametes travel during their lifetimes, which sets the scale at which populations vary (\cite{cain2000long, kremer2012long}).
Of particular interest is both the average distance travelled and the shape of the dispersal distribution \citep{cain2000long, nathan2012dispersal}.
In plants, dispersal is often characterised by leptokurtic or 'fat-tailed' distributions, where dispersal is most likely to occur over short distances, but there is a long tail of long-range dispersal events (\cite{clark1998trees,austerlitz2004using,bullock2017synthesis}).
This leptokurtosis allows much more rapid dispersal than would be suggested by the average dispersal distance alone, with long-range migrants having a disproportionate effect on the spread of adaptive alleles (\cite{clark1998trees,cain2000long}).
We thus aim to reconstruct mating events between individuals to infer the distribution of dispersal.

Pedigrees are inferred using three sources of information.
First, the oldest and most common approach is to identify parent-offspring relationships that are most likely given on Mendelian inheritance, treating offspring individuals separately \citep{thompson1976inference, meagher1986analysis, marshall1998statistical}.
Second, sibling-reconstruction approaches use information about alleles shared between siblings to jointly infer sibling relationships with the parentage of the entire sibship (e.g. \cite{thompson1987parental, emery2001assignment, thomas2002sibship, jones2007estimating, wang2004sibship, anderson2016bayesian, huisman2017pedigree}).
Because sibling relationships are typically not known and there are a very large number of possible configurations to consider, a major challenge in sibship inference is to explore this space effectively.
Genetic information shared between parents, offspring and between siblings are vital for reconstructing their relationships.

Finally, mating patterns and differences in mating success often depend on additional population and phenotypic parameters that are not related to genetics.
Jointly modelling these processes with pedigree relationships can not only better resolve relationships between individuals, but provide more accurate estimates of population and phenotypic factors compared with post-hoc estimates. \citep{neff2001bayesian, hadfield2006towards}.
Pollen dispersal is a good example of this.
We expect nearby individuals to mate with one another more often than with more distant individuals.
By using a suitable model for how the probability of mating decays with distance we expect to weight inference to more likely candidate mates, but also to directly infer meaningful parameters about pollen flow \citep{adams1992using, jones2003maximum, oddou2005pollen, klein2008pollen, chybicki2013seeing}.
A particularly well-established approach for this the the 'neighbourhood model' which jointly estimates parent-offspring relationships and dispersal distributions whilst also accounting for pollen immigration, incomplete sampling, variation in fertility and assortative mating \citep{adams1989mating, burczyk2004cautions, oddou2005pollen, chybicki2013seeing, gerard2006assortative, chybicki2018nmpi, chybicki2021identification}.
Jointly modelling parent-offspring relationships with population parameters has great appeal because it allows us to directly address biologically relevant relationships between traits of interest and mating patterns.

Approaches exist for combining parentage with sibship information, and parentage with population parameters, and this additional information is expected to increase the accuracy of pedigree reconstruction \citep{neff2001bayesian, wang2007parentage, walling2010comparing}.
However we currently lack a framework for utilising all three sources of information - parentage, sibships and population parameters - in a single joint analysis.
Because of this, we also lack an understanding of the relative contribution of sibship and population parameters to pedigree inference, and how this changes depending on the data available and the biology of the system.
We previously described a package Fractional Analysis of Paternity and Sibships (FAPS) to jointly infer of paternity and sibships (\cite{ellis2018efficient}).
In this paper we describe how to include non-genetic information into the FAPS procedure, and use MCMC to update parameters for those data to jointly infer paternity, sibships and population parameters.
We then apply this to the inference of the pollen dispersal distribution in a hybrid-zone population of the snapdragon \textit{Antirrhinum majus} with a large number of potential pollen donors.
Using simulations we also characterise how the power to reconstruct paternity of individuals and mating between adults depends on the information available from genetics markers, the degree of shared paternity and the scale of pollen dispersal.

%------------------------------------------------

\section{Materials and Methods}

\subsection{\textit{A. majus} data}

\subsubsection{Study population}

\begin{figure*}
    \centering
	\includegraphics[]{fig-map.eps} % Figure image
	\caption{
        Map of the hybrid zone.
        The map shows the distribution of maternal plants, inferred pollen donors and remaining plants along the lower (South) and upper (North) roads. Only pollen donors for families with one or more offspring are shown.}
	\label{fig:map} % Label for referencing with \ref{bear}
\end{figure*}

We examine a hybrid zone population of the snapdragon \textit{Antirrhinum majus} ssp \textit{majus} in the Spanish Pyrenees. Here, the yellow-flowered \textit{A. m. majus} var \textit{striatum} and the magenta-flowered \textit{A. m. majus} var \textit{pseudomajus} meet and hybridise to produce diverse recombinant phenotypes, including pink, white and orange flowers \citep{whibley2004molecular}.
The population grows along two parallel roads running East-West close to Rib\`{e}s de Freser (figure \ref{fig:map}). The ’lower’ road is at 1150-1200m above sea level, whilst the ’upper’ road climbs 1250-1500m, and is 500-1000m north of the lower road. Hybrids are mostly confined to a 1km ’core’ hybrid zone, with \textit{A. m. striatum}- and \textit{A. m. pseudomajus}-like plants becoming dominant to the West and East respectively. We surveyed as many flowering plants as we could find in June and July of 2012 (n=2124), and collected information on flower number and location using a Trimble GeoXT datalogger. We collected two to three leaves for DNA extraction and dried these in silica gel (Fischer Scientific). We also sampled one mature flower from each plant and scored this for yellow and magenta pigmentation as described by \cite{whibley2004molecular}. \textit{A. majus} grows in disturbed habitats such as roadsides and railways; they are rare in the established forest and pasture between the two roads, on the north-facing slope to the South, and on the high mountain peak to the North of the two roads (figure \ref{fig:map}). It thus is likely that we sampled the majority of the plants that flowered during the study period, although we cannot exclude that some flowering plants may have been missed. 

Pollination is carried out exclusively by large bumblebees and carpenter bees who are large enough to open the flowers (\cite{vargas2010occluded, andalo2019prevalence}). \textit{A. majus} has a gametophytic self-incompatibility system, and self-pollinated seeds are very rare \parencite{surendranadh2022effects}. There are no detectable post-zygotic barriers between \textit{A. m. striatum} and \textit{A. m. pseudomajus} (\cite{andalo2010post}).

In August 2012 we collected a single, mature, wild-pollinated fruit from each of 60 mothers (figure \ref{fig:map}). In order to minimise disturbance to the population we only sampled from plants which had set a minimum of five mature fruits. These mothers were chosen to represent an even sample of pigmentation phenotypes, spread as evenly as possible across the core of the hybrid zone where hybrids are most dense, resulting in 10 \textit{A. m. striatum}-like, 17 \textit{A. m. pseudomajus}-like, and 33 hybrid-phenotype mothers.

\subsubsection{Genotyping}

We grew seeds in 5cm plug trays filled with potting compost (Gramaflor) in a greenhouse under Sylvania GroLux lights on a 16-hour cycle. We sowed three seeds per plug for 50-70 plugs per maternal family and thinned seedlings to a single seedling per plug after cotelydons had appeared. We transferred approximately 1cm$^2$ of fresh tissue from 1419 seedlings to 96-well DNA-extraction plates (LGC Genomics, Berlin) and allowed tissue to dry using the sample bag and silica gel provided. For parental tissue from the hybrid zone we transferred approximately 1cm$^2$ tissue dried in the field to the same plates. DNA extractions of the plated tissue samples were carried out by LGC Genomics.

We genotyped tissue samples at 71 SNPs by KASPR sequencing (LGC Genomics).
These SNPs are a subsample of a panel used for a wider survey of the hybrid zone (\cite{surendranadh2022effects}).
Previous work identified the per-locus genotyping error rate in these data to be approximately $10^-4$ (\cite{surendranadh2022effects}).
We removed 110 offspring that had missing data at more than 20\% of the SNPs.
We also pruned 4 SNPs that showed more than 20\% missing data, or less than 15\% heterozygosity.
This left us with a set of 1308 offspring from 60 maternal families and 2124 candidate pollen donors, with between three and 39 offspring per maternal family (mean=21.8), with 64 marker SNPs.

\subsection{Joint estimation of paternity, sibships and dispersal}

\subsubsection{Probability model}

We begin with observed SNP marker data \textbf{M} for mothers, offspring and candidate father, and a matrix \textbf{D} of Euclidean distances between mothers and all possible candidate fathers, and the per-locus genotyping error rate $\epsilon$. From this we wish to infer pedigree P describing sibling, paternal and (known) maternal relationships, and vector $\theta$ of dispersal parameters. The full probability model is then
\begin{equation}
\label{eqn:probability_model}
\begin{split}
\Pr( P, \theta | \textbf{M}, \textbf{D}, \epsilon) \propto \Pr(\textbf{M} | P, \textbf{D}, \theta, \epsilon)
\Pr(\textbf{D} | \theta, P) \\
\Pr(\theta) \Pr(P) \Pr(\epsilon)
\end{split}
\end{equation}

In the following sections we outline how to extend our existing method for inferring sibships and paternity to include data from non-genetic covariates, a model for pollen dispersal, suitable hyperprior distributions for dispersal parameters, and a procedure to infer the posterior distribution of the parameters of interest.

\subsubsection{Allowing for covariates in paternity inference}

We have previously described a Python package FAPS which performs joint analysis of paternity and sibship relationships for sibling arrays based on SNP data, and allows for integrating out uncertainty in relationships (\cite{ellis2018efficient}). Here we extended the software to allow for additional non-genetic information to be included.

FAPS begins with marker data for one or more maternal families composed of a mixture of half- and full-siblings, the known mother of each family, and an array of candidate fathers.
\textcite{ellis2018efficient} described a method to infer sibling and paternity relationships in those individuals; we refer the reader to that paper for the full details, but review the most important aspects here.
FAPS uses marker data \textbf{M} to build matrix \textbf{G} of probabilities of paternity based on Mendelian transition probabilities for each maternal family.
\textbf{G} has a row for each offspring and a column for each candidate father, where element $g_{ij}$ is the probability that candidate \textit{j} is the father of offspring \textit{i} given marker data, and an estimate of genotyping error rate $\epsilon$. 
The final column of \textbf{G} is the probability that the father of each offspring was missing from the sample of candidates, based on the probability of observing offspring alleles from population allele frequencies.
Rows in \textbf{G} sum to one, and describe a multinomial distribution of probabilities of paternity over all candidate fathers, including unsampled fathers.
FAPS then builds a similarity matrix whose \textit{ih}\textsuperscript{th} element is the likelihood

\begin{equation}\label{eqn:faps_similarity_matrix}
    \Sigma_j^F g_{ij}g_{hj}
\end{equation}

that the \textit{i}\textsuperscript{th} and \textit{h}\textsuperscript{th} offspring are full siblings by summing over the probabilities that they share any one of the $F$ candidate fathers. The similarity matrix is used to perform hierarchical clustering to identify plausible ways to partition offspring into families of possible full sibships, and the likelihood of each partition structure is estimated by Monte-Carlo simulation. The likelihood that candidate father \textit{j} is the father of putative full sibship \textit{k} is then
\begin{equation}\label{eqn:faps_lik_sibship}
\frac{
    \Pi_i^n g_{ij}
    }{
    \Sigma_j^F\Pi_i^n g_{ij}
    }
\end{equation}
for all \textit{n} offspring in \textit{k}. FAPS returns a vector of probabilities (which sum to one) that each candidate is the father of sibship \textit{k}, or that the true father was not sampled. This is done for each plausible partition structure. This gives a distribution of possible partition structures and their likelihoods, and a distribution of paternity probabilities for each putative family within each partition structure.

We can incorporate non-genetic information about paternity using a suitable function relating those data into probabilities of paternity \citep{hadfield2006towards}.
In principle this can be done for any kind of data for which there is a suitable function relating the observed data for a set of candidate fathers to a probability of having mating with each mother, given a set of parameter values for that function. The goal is to find parameter values that best explain the data. For example, a standardised continuous phenotype $z$ could be modelled with the logistic function $1/(1+e^{\beta z})$, where $\beta$ describes the relationship between $z$ and male fertility.
A categorical phenotype can be modelled as a multinomial vector of probabilities that sum to one.
In this study we use matrix \textbf{D} of Euclidean distances between mothers and candidate fathers as a covariate, and model the probability $\Pr(d_{mj}|\theta)$ of a mating event occurring between mother \textit{m} and candidate \textit{j} who are distance $d_{mj}$ apart, based on a suitable model of pollen dispersal (see below).
Assuming covariates of mating patterns are independent of marker genotypes it is then straightforward to incorporate this into the procedure outlined above by modifying eqn. \ref{eqn:faps_similarity_matrix} to
\begin{equation}\label{eqn:pairwise_with_covariates}
\Sigma_j^F \Pr(d_{mj} | \theta)g_{ij}g_{hj}
\end{equation}
and eqn. \ref{eqn:faps_lik_sibship} to
\begin{equation}
\label{sibship_with_covariates}
\frac{
    \Pr(d_{mj} | \theta)\Pi_i^n g_{ij}
    }{
    \Sigma_j^F \Pr(d_{mj} | \theta)\Pi_i^n g_{ij}
    }
\end{equation}

Eqn. \ref{sibship_with_covariates} can then be used to estimate the likelihood of the partition structure by Monte-Carlo simulation, as previously described by \textcite{ellis2018efficient}.
For a given set of dispersal parameters in $\theta$ the likelihood of the whole maternal family is then the sum of likelihoods for each possible partition. When there are multiple maternal families, the likelihood of the whole dataset given $\theta$ is the product of those likelihoods over each maternal family. When we update $\theta$, this likelihood will change, giving us a way to compare likelihoods of different dispersal parameter values and identify those most consistent with the data.

\subsubsection{Pollen dispersal distribution}

We wish to describe the distribution of distances travelled by pollen from anther to stigma \citep{nathan2012dispersal}.
A useful function for describing plant dispersal distributions is the generalised normal distribution (\cite{clark1998trees,Nadarajah2005,kremer2012long}). This is a generalisation of the exponential family of probability distributions and includes the exponential and standard normal distributions as special cases, but allows for fat and thin tails. It is commonly used to model plant dispersal distributions because these are often found to show clear kurtosis (e.g. \cite{austerlitz2004using, robledo2005patterns, klein2008pollen, burczyk2019patterns, field2011importance, ottewell2012pollen}). The generalised normal distribution describes the probability of observing dispersal distance $d_{mj}$ between mother $m$ and candidate father $j$ given scale parameter $a$ and shape parameter $b$ as:
\begin{equation}
    \Pr(d_{mj}|a,b) = \frac{1}{K} \exp{  (-|\frac{d_{mj}}{a}| ^b) }
    \label{eqn:GND}
\end{equation}
where
\begin{equation}
    K = \Sigma_j^F \exp  (-|\frac{d_{mj}}{a}| ^b) 
\end{equation}
The function takes the forms of the standard exponential when $b=1$ and normal distribution when $b=2$. Values of $b<1$ reflect leptokurtic distributions with tails that decay more slowly than would be expected under the exponential distribution. Using $\theta={a,b}$, this provides a convenient function for calculating $\Pr(d_{mj} | \theta)$ because it allows for a long tail of long-distance migrants.
Scaling by $K$ ensures that probabilities of dispersal distances across fathers sum to one.

Sometimes it may be that an unrelated candidate has a similar or higher probability of paternity of one or more offspring simply due to stochasticity in Mendelian sampling.
Whereas true fathers are expected to be (on average) close to the mother, other candidates should be drawn at random from the population (although this is complicated if there is substantial spatial genetic structure).
This will inflate the apparent kurtosis in the data and bias $b$ downwards.
To accommodate this we modify \ref{eqn:GND} to model dispersal as a mixture of a generalised-normal and a uniform distribution:
\begin{equation}\label{eqn:mixture_model}
\Pr(d_{mj} | a,b,\lambda) = \frac{
    \lambda \Pr(d_{mj} | a,b) + \frac{1-\lambda}{F}
    }{
    \Sigma_j^F( \lambda \Pr(d_{mj} | a,b) + \frac{1-\lambda}{F})
    }
\end{equation}
where $F$ is the number of candidate fathers and $\lambda$ is a mixture parameter determining the proportion of the probability mass due to ’real’ dispersal.
The uniform part of this mixture allows for signal coming from incorrect candidates without requiring the 'true' dispersal kernel to be unnecessarily leptokurtotic.
This approach is conceptually similar to those of \cite{streiff1999pollen} and \cite{slavov2009extensive} who modelled dispersal as a mixture of distinct processes describing short- and long-range dispersal up to a maximum distance estimated from the data.

It is common to also report the mean dispersal distance as the square root of the variance of this distribution as
\begin{equation}
\label{eqn:sd_GND}    
\sqrt{ \frac{ a^2 \Gamma(3/b) }{ \Gamma(1/b) } }
\end{equation}
(\cite{Nadarajah2005}).
We prefer to focus on median dispersal distances because this is more intuitive for highly skewed distributions, and also to estimate this on realised inter-mate distances rather than on parameters alone (see "Inference of mating patterns"). We nevertheless report this and how it relates to the effect of $\lambda$ on results.

\subsubsection{Priors for dispersal parameters}

We require prior distributions for dispersal parameters $a$, $b$ and $\lambda$.
We used a log-normal prior for $b$ ($\ln b \sim N(\mu=0, \sigma = 0.5)$.
This model allows for a range of shape values reflecting leptokurtosis to Gaussian dispersal, but is weights against very strong over- or under-dispersion in the dispersal distribution because prior probabilities approach zero as $b$ approaches 0 or 3 (figure S1).
We used a Gamma prior for $a$ because this describes positive continuous values, and because it is conjugate with the variance parameter of a Gaussian distribution.
Because the effect of $a$ depends on $b$ and has no intuitive biological interpretation itself, we used prior simulations to choose a suitable parameterisation for $a$. We first simulated 10000 pairs of values for $a$ and $b$ from Gamma and log-normal distributions respectively. We then used each pair of simulated values to parameterise a generalised normal distribution, and simulated 1000 dispersal distances from that distribution, and visually examined the distribution for biological plausibility. We chose a Gamma distribution with shape=6 and scale=50, because this gave dispersal distributions with most dispersal occurring within 500m, but allowing for rare long-range dispersal events (figure S1).

For $\lambda$, we use a beta distribution with parameters Beta(1.1, 1.1).
This distribution approaches zero when $\lambda$ is close to zero or one, but is fairly flat in between.
This implies that we do not expect that all the weight should be on either the generalised-normal or the uniform components of the mixture distribution in eqn. \ref{eqn:mixture_model}, but that we do not have strong prior beliefs about values between that.
To examine the effect of modelling dispersal as a mixture model at all, we also repeated the MCMC with $\lambda$ to set to one.

\subsection{Inference}

\subsubsection{Inference via MCMC}

We used the Metropolis-Hastings Markov-chain Monte Carlo algorithm to infer the posterior distribution of dispersal parameters $a$, $b$ and $\lambda$.
We ran four independent chains beginning from distant areas of the parameter space.
At each iteration, we perturbed the value of each parameter by a factor drawn from a normal distribution with a fixed standard deviation for each parameter ($\sigma= 2$ for $a$; $\sigma= 0.05$ for $b$; $\sigma= 0.025$ for $\lambda$).
We ran each chain for 25500 iterations and subsequently removed the first 500 iterations of each as burn-in.
After checking chain convergence we thinned subsequent iterations to retain 250 posterior draws from each chain for further analyses, for a total of 1000 posterior draws.

\subsubsection{Inference of mating events}

We aimed to create a list of possible mating events between mothers and candidate fathers consistent with the data to identify a set of independent pollen dispersal events.
FAPS generates one or more partition structures that a half-sibling family could be split into full-sibling families.
For each partition structure, FAPS identifies valid set of distinct fathers for each full sibship, and estimates a likelihood for each father-sibship configuration.
Each set of fathers reflects a hypothetical set of mating events, and the probability that a single father mated with the maternal plant is the sum of probabilities for each partition structure in which he sires at least one offspring.
This gives a list of mating events, with a posterior probability that each occurred, including an entry for offspring whose father was missing.
The posterior estimate of offspring number is the number of offspring for each father in each father-sibship configuration weighted by the probability of the mating event.
Note that the number and size of sibships are not necessarily integers because estimates are weighted averages over plausible sibship partition structures.
We inferred mating events in this way using eqn. \ref{eqn:faps_lik_sibship} based on dispersal parameter estimates from each of 1000 iterations of the MCMC output.

We previously noted that when FAPS is incorrect, it tends to split one individual in a sibship into a singleton family, but essentially never incorrectly merges unrelated sibships \cite{ellis2018efficient}.
We aimed to identify correct mating events based on family sizes.
Previous investigations of FAPS showed that the main source of error was individuals from a larger sibship group being falsely assigned to a singleton sibship, although true half-siblings are almost never merged as full siblings (\cite{ellis2018efficient}).
This predicts that those singleton families will have posterior-mean family sizes of less than one because the nominal number of offspring is one, but the posterior probability for the family is less than one.
This idea is not necessarily intuitive, so to illustrate this, imagine three offspring (A, B and C) with two candidate fathers (X and Y) and that there are two plausible paternal-sibships configurations,: (1) X is the father of A, B and C, with probability 0.6, or (2) X is the father of A and B, while Y is the father of C with probability 0.4.
The probability that X mated with the mother is 0.6+0.4=1, but the probability that Y mated with the mother is 0.4.
Weighted mean offspring numbers are (3*0.6)+(2*0.4)=2.4 for X and 1*0.5=0.5 for Y.
We can be more confident that the mother mated with X than with Y whilst being agnostic about the paternity of individuals.
Based on this idea, we distinguish between mating events with posterior-mean family sizes greater or equal to one, and less than one.

Finally, we estimated the relationship between the number of offspring in a maternal array and the number of paternal families therein.
We first averaged paternal family size for each mother-father pair over iterations of the MCMC, then fitted a Poisson generalised linear model (GLM) of paternal family size on maternal family size (\cite{McCullagh1989}).

\subsection{Simulations}

We used simulations to determine the statistical power of the dataset and and how this depends on the information used.

We simulated mating events between each true mother and one or more sires drawn from the pool of 2124 genotyped adult plants.
Each maternal plant mated with the same number of pollen donors as inferred in the empirical analysis to generate one or more full-sib families per mother.
For simplicity we drew sires for each family based on the probability of mating with a mother given an exponential pollen dispersal distribution (i.e. $b=1$ and different values for $a$; see below).
We ignored mating events with posterior probabilities less than 0.9.
We simulated offspring genotypes based on Mendelian segregation and added genotyping errors to adults and offspring genotypes with the observed genotype error rate of $10^{-4}$ per locus per individual.
In this way simulated data reflects the the observed genetic and spatial structure of the \textit{A.~majus} hybrid zone.

We parameterised simulations by varying four parameters that are likely to affect inference of relationships.
We aimed to use values that are likely to affect cases where parameters are less informative, moderately informative, and more informative.
To vary marker information we downsampled the number of genotyped markers to 40, 50 and 60.
To vary the information about paternity shared between full siblings we varied he number of offspring sired by each father (1, 3 and 5 offspring)
To simulate missing fathers we randomly selected 10\%, 30\% and 50\% of true sires and removed them from the set of candidate fathers.
We simulated dispersal with mean distances of 3m, 30m and 300m.
We repeated each parameter combination 100 times.
We note that it is now routine to generate data with many more markers than used here, and in a previous study we did simulate additional SNPs \citep{ellis2018efficient}.
However, denser markers will be in stronger linkage disequilibrium, meaning they are no longer independent and existing likelihood calculations are no longer valid.
As such, additional markers do not necessarily translate into increased statistical power in a straightforward way.
Using the observed markers also allowed us to capture the genetic structure of the population.

We used simulated datasets to infer mating events using the same procedure as for the empirical dataset.
We compared results based on mating events inferred using three sources of information.
First, we used most likely mother-father-offspring trios inferred using Mendelian transition likelihoods ('paternity only').
Second, we inferred mating events using information about both paternity and sibling relationships (as described by \cite{ellis2018efficient})
Third, we combined information from paternity, sibling relationships and the shape and scale of the dispersal distribution together (this paper).
To test the hypothesis that we can distinguish between reliable and dubious mating events based on family size we quantified the probability that an inferred mating event was correct separately for mating events with posterior-mean family sizes of one or more, or less than one.
To investigate different datasets and analyses estimated real dispersal we compared true realised median dispersal distances in each simulated dataset to median dispersal from inferred mating events.

\section{Results}

\subsection{Family sizes vary in \textit{A.~majus}}

In the empirical \textit{A.~majus} dataset we identified an average of 195 full-sibling families for which a father could be positively identified across MCMC samples (96\% credible intervals (CI): 189, 199).
6.5\% families had posterior probabilities and posterior-mean family sizes less than one, showed substantial uncertainty across MCMC iterations (figures S2) and seem to be drawn at random from across the population (figure S3), indicating that these mating events are likely to be dubious.
In contrast, the remaining 168 families showed strong posterior support with little variation between MCMC iterations (figure S2), and were clustered closer to the maternal plants (figure S3).
These likely reflect robust independent mating events that can be used to infer a dispersal kernel, and we focus on these mating events in the rest of this manuscript.

Full-sibship sizes ranged from one to 18 (figure S4).
The GLM of the number of full sibships on maternal-family size revealed that (natural) log number of sibships increased by 0.045 for every additional offspring included in the maternal family (intercept~=~0.40; figure S5). This corresponds to detecting a new family for approximately every 4 or 5 offspring genotyped, on average.

For all sixty mothers we found evidence for paternity by one or more unsampled fathers with non-zero probability (figure S2).
These families include an average of 789 offspring (CIs: 785, 793).
This corresponds to around 60\% of the offspring.

\subsection{Pollen dispersal is leptokurtic and asymmetric}

\begin{figure*}
    \centering
    \includegraphics{fig-posterior_distributions.eps}
    \caption{
        Posterior densities for the shape, scale, and mixture parameters of the dispersal kernel.
        Histograms show stacked densities for each of four independent chains.
    }
    \label{fig:posterior_summaries}
\end{figure*}

\begin{figure*}
    \centering
    \includegraphics{fig-dispersal.eps}
    \caption{
        Distribution of pollen-dispersal distances.
        (A) Number of mating events in 30m bins.
        (B) Cumulative distribution of distances for dispersal events for paternal families with one or more offspring.
        Note the log$_{10}$ scale of the x-axis in B.
        Mating events are weight weighted by their posterior probability.
        The histogram is summed over iterations of the MCMC.
        Separate cumulative curves are shown for 1000 MCMC iterations separately.
    }
    \label{fig:dispersal}
\end{figure*}

The posterior distribution for the dispersal shape parameter was consistently less than one, with a posterior mean of 0.57 (96\% CIs: 0.46, 0.70), indicating a leptokurtic dispersal kernel (figure \ref{fig:posterior_summaries}A).
Posterior mean scale was 22.0 (95\% CIs: 9.1, 37.4; \ref{fig:posterior_summaries}B).

Consistent with this, the distribution of pollen dispersal distances for families with one or more offspring shows a peak close to zero and a long tail of long-distance dispersal events up to 859m (figure \ref{fig:dispersal}).
Mean and median dispersal distances were 83.4 and 28.0 metres respectively.
With two exceptions, all mating occurred between plants on the lower road (figure \ref{fig:map}).
The cumulative distribution of realised dispersal distances varied very little across the posterior distributions of scale and shape parameters (figure \ref{fig:dispersal}B), indicating that the estimated dispersal distribution is robust.
These results indicate a leptokurtic pollen dispersal kernel.

\subsection{Mixture parameter influences dispersal shape}

The posterior distribution the mixture parameter $\lambda$ was centred around 0.97 (96\% CIs: 0.89, 1.00; figure \ref{fig:posterior_summaries}C).
This indicates that up to 11\% of paternal families are compatible with non-sires located at medium to long distances from the mother.
This may be because the true father is missing and/or stochasticity in Mendelian sampling of the genotypes.
If ignored, this signal would inflate apparent leptokurtosis in the dataset.
Consistent with this, when we constrain $\lambda$ to be fixed at 1 the posterior distributions for the scale and shape parameters are shifted downwards compared to the analysis where $\lambda$ is allowed to vary (posterior means for $a$ and $b$ drop from 22.0 to 15.0 and from 5.7 to 4.7 respectively; figure S6).
This would indicate more strongly leptokurtic pollen dispersal (figure \ref{fig:posterior_summaries}A, \ref{fig:posterior_summaries}B).
When mean dispersal distance is estimated from the second moment of the generalised normal distribution (eqn. \ref{eqn:sd_GND}; \cite{clark1998trees}) this would imply a substantial increase in mean dispersal from 137 to 227 metres (figure S7).
In this instance then, including a mixture component in the dispersal kernel reduces apparent mean dispersal distance when this is inferred from the moments of the generalised-normal dispersal distribution.

\subsection{Sibship and population information improve pedigree and dispersal inference}

\begin{figure*}
    \centering
    \includegraphics{fig-sim_mating.eps}
    \caption{
        Mating events in simulated data.
        Values show the probability that inferred mating events are correct for families with posterior-mean family sizes of (A) more or more or (B) less than one.
        Panels show the proportion of true fathers missing from the dataset used to infer mating events (columns) and the true full sibship size (rows).
        Values are the mean of 100 replicate simulations, and are shown for inference using paternity (Pat) only, paternity and sibships (sibs), and paternity, sibships and dispersal information.
        Note that values for inference with paternity only are not shown in A because these genealogies are inferred from the most-likely configuration, so a posterior-mean family size is not defined.
    }
    \label{fig:sim-mating}
\end{figure*}

We used simulations to investigate how well we can distinguish true and false mating events based on posterior-mean family size.
When families contained only a single offspring a large proportion of correct mating events had dubious support and posterior-mean family size less than one (figure \ref{fig:sim-mating}A).
This was more pronounced with increased numbers of markers, and when dispersal information was included in the analysis.
When real family sizes were greater than one however, almost no true mating events had posterior-mean family sizes less than one.
This means that when families contain multiple siblings there is no cost to ignoring apparent singleton families.
Distinguishing true and false mating events based on family size gives mixed results for singleton families, but is an effective means to identify true mating events for larger families.

The simulations also reveal how well we can reconstruct mating events under different scenarios and using different data.
Inferred mating events were more likely to be correct as the number of marker loci and the proportion of true fathers that were sampled increased, and decreased with larger families (figure \ref{fig:sim-mating}B).
We found small differences in accuracy between dispersal scenarios, but the direction depended on what information was used to infer mating events.
For example, when only paternity trios were used, the least informative scenario (mean dispersal = 300m) showed the highest accuracy, and the most informative (mean dispersal = 3m) the lowest, but when information from paternity and sibships were combined the opposite pattern was true.
The clearest result is that there was a substantial gain in power to detect mating events when paternity and sibship information are used raelative to paternity only, and when paternity, sibship and dispersal information are used relative to paternity and sibships only.
Including information from both sibships and population parameters greatly increase the accuracy of inferred mating events.

A substantial proportion of simulated offspring were assigned to unsampled fathers (figure S8).
Interestingly, this increased with increasing number of markers, indicating that FAPS becomes more conservative about assigning paternity to sampled fathers as marker information increases.
We found that models using only paternity, or paternity and sibship information tended to underestimate the true proportion of offspring whose father was missing when that proportion was high, but were largely correct when that proportion was low.
In contrast, the model that included dispersal information tended to overestimate the true proportion of offspring whose father was missing when that proportion was low, but were largely correct when that proportion was high.
The improvement in identifying true mating events when including additional information may come with a cost of increased false-negative assignment to unsampeld fathers in some circumstances.

\begin{figure*}
    \centering
    \includegraphics{fig-sim_dipsersal.eps}
    \caption{
        Errors in dispersal estimates in simulated data.
        Values show the absolute difference in median dispersal events between real mating events and inferred mating events with posterior-mean family sizes of one or more.
        Panels show the proportion of true fathers missing from the dataset used to infer mating events (columns) and the true full sibship size (rows).
        Values are the mean of 100 replicate simulations, and are shown for inference using paternity (Pat) only, paternity and sibships (sibs), and paternity, sibships and dispersal information.
    }
    \label{fig:sim-dispersal}
\end{figure*}

We compared median dispersal of inferred mating events to true median dispersal (figure \ref{fig:sim-dispersal}).
Dispersal estimates were substantially more accurate when more markers were used, but showed less of a dependence on family size or the proportion of sampled fathers.
Instead, estimated dispersal distances depended strongly on true dispersal distance; estimates were most accurate for datasets with the shortest dispersal and lest accurate for dispersal over the largest distance.
Estimated dispersal also depended strongly on the information used to infer mating events.
Analyses using paternity information only often overestimated dispersal by more than a hundred metres.
Including information about sibling relationships reduced this estimate substantially, and including information about dispersal reduced these even further.
Of particular note is that when all 67 markers are used, as in our empirical dataset, the estimate of median dispersal is typically within one metre of the true value, indicating that our empirical estimate is robust.
Inference of dispersal distance is most reliable when dispersal is over relatively short scales, and as much information as possible is included in the analysis.

\section{Discussion}

We have described a framework to jointly infer paternity, sibling relationships and population parameters. 
Applying this to a natural populations of snapdragons, we find strong evidence for a leptokurtic pollen-dispersal distribution.
Using simulations we demonstrate that sibship and population information improves inference of genealogical relationships and biological conclusions.

\subsection{Implications for the hybrid-zone population}

The shape of the dispersal kernel implies that most mating occurs within tens of metres, but that there is a long tail of mating events between individuals at up to several hundred metres. This result suggests that any natural selection acting through male reproductive components (pollen dispersal) occurs at small spatial scales. If pollinators discriminate between different flower-colour phenotypes in the hybrid zone, this sets an important geographic scale of selection for empirical tests on pollinator-mediated selection. For example, within 20m in the core of the hybrid zone, pollinators would encounter a full range of colour phenotypes that would provide scope for colour discrimination and differential visitation to influence male and female fitness. In this paper we have not attempted to estimate selection via flower colour because the number of mothers for each flower colour is small, meaning that we likely have little statistical power. However, in ongoing work we are investigating variation in fitness in a much larger panel of offspring arrays, which will allow us to address such questions with much greater accuracy.

Rare long-distance dispersal events can have unexpectedly large effects on natural populations \citep{clark1998trees}.
We found that the most extreme pollen-dispersal events spanned many hundreds of metres.
Since the width of the core of the hybrid zone is approximately one kilometre, this means that pollen would be able to quickly move across a substantial fraction of hybrid zone, and accelerate introgression.
However, there are several reasons why long-distance dispersal events are especially difficult to detect.
First, these events are intrinsically rare.
Second, as distance increases the more plants need to be sampled to ensure the father is included.
Third, as more candidate fathers are included the greater the chance that an unrelated fathers is falsely assigned.
It is thus likely that we have under-estimated the tail of long-distance dispersal events to some extent.
However, because this population is mostly confined to roadsides, our population sample extends several kilometres further than the longest dispersal event detected, and we can minimise false-positive paternity assignments by conditioning on family size, this underestimate is unlikely to be substantial.
Nevertheless, it remains possible that even more long-range dispersal events may occur in this population, for example by larger queen bumblebees early in the season \citep{lepais2010estimation}, with the potential to have a large impact on gene flow.

\subsection{Controlling bias in dispersal estimates}

\subsubsection{Bias due to incorrect fathers}

Sometimes candidate pollen donors can have a similar or greater Mendelian probability of paternity for an offspring individual as does the true father due to stochasticity in Mendelian sampling (\cite{thompson1976paradox}).
If not addressed, we might infer an incorrect mating event between adult plants, which would bias estimates of dispersal upwards.
Two aspects of our study suggest that signal from false fathers should have a minimal effect on our results.

First, our simulations showed that incorrect families could be readily identified based on family size.
In previous simulations to test the performance of FAPS we found that the main source of error was individuals from a larger sibship group being falsely assigned to a singleton sibship, but this analysis focussed on the most likely partition structure to simplify results (\cite{ellis2018efficient}).
In this study we found that those incorrect singleton families have low posterior probabilities when considered as part of a whole set of plausible sibship configurations, meaning that family sizes are effectively less than one individual.
Since our goal was to infer pollen dispersal by identifying mate pairs, excluding mating events for families with sizes less than one is an effective way to identify a conservative set of high-confidence mating events, even if this means that some true mating events are lost.
Consistent with this, in simulated data that closely match the \textit{A.~majus} dataset we found that estimated dispersal was very close to the true values, indicating that false-positive mating events have not had a large influence on our estimates of dispersal.
This demonstrates that accounting for uncertainty in sibship structure is an extremely useful way to improve the accuracy of paternity inference.

Second, we modelled dispersal as a mixture distribution with a term allowing for false-positive fathers to try and reduce the bias in estimates of dispersal.
The mixture distribution reduces the bias in the estimated shape parameter and the second moment of the generalised normal distribution, often interpreted as mean (squared) dispersal distance (e.g. \cite{clark1998trees, austerlitz2004using, klein2008pollen}).
We note that this bias will exist to some extent as long as knowledge of paternity is less than perfect.
Moreover, two factors may complicate the interpretation of the mixture paramter.
First, the idea assumes that unrelated candidates are distributed at random with respect to genotypes, which will not be the case if there is spatial genetic structure.
In the case of the \textit{A.~majus} populaton the marker panel was designed to show very little spatial structure, so this is unlikely to have a substantial impact.
Second, more distant mating events may be real, but reflect some other distribution process than that modelled by the generalised normal distribution.
In this case, the mixture component is still useful in that it allows these two distributions to be decoupled, at least to some extent \citep{streiff1999pollen, slavov2009extensive}.
Here, our biological conclusions are not affected, because we focus on the distribution of realised mating events, which are affected in only a minor way in real terms (figure \ref{fig:dispersal}).
However, caution is warranted in the interpretation of the raw parameter values for shape and mean dispersal from the generalised normal distribution (figure S8).
It is possible that the effect was stronger here than would be the case in other published studies because our sample of candidate fathers is large, and the spatial extend is much larger (two orders of magnitude) than median dispersal distance.
In contrast, other studies have focussed on wind-pollinated trees, where pollen can travel much further, and had smaller numbers of candidates (e.g. \cite{adams1992using, austerlitz2004using, klein2008pollen}).
This demonstrates the utility of using inferred dispersal parameters to inform inference of real mating events, rather than focussing on the parameters themselves.

\subsubsection{Missing fathers}

An alternative source of bias comes from false negative paternity, where the true father of a family is present but is inferred to be missing.
We found that more than half the offspring were inferred to have an unsampled father.
Since these are cases were paternity has not been falsely assigned to an unrelated candidate, we do not expect false assignment to bias dispersal estimates upwards.
Nevertheless, the extent to which these missing fathers bias dispersal estimates still depends on their distribution in space.
At the same time, it may be that sampling effort was greater in the core of the hybrid zone than at the edges of our sampling region.
It is known that bumblebees, especially queens, can disperse over several kilometres, so very long range pollen dispersal is plausible (\cite{osborne2008bumblebee, hagen2011space, lepais2010estimation}).
In this way it is possible that we have missed long-distance mating events, which would bias dispersal estimates downward.
However, our sample of candidate fathers is large, and extends at least a kilometer beyond the range of plants identified as pollen donors (figure \ref{fig:map}), which is more than twice as far as the longest dispersal event detected (figure \ref{fig:dispersal}).
We feel it is thus unlikely that there could have been enough unsampled fathers far from the sample of mothers to greatly influence dispersal shape or median dispersal distance.

An alternative explanation for the high number of missing fathers is that they finished flowering before the main sampling began.
This is a more plausible explanation, because the population had already begun to flower at the time of the first surveys in June 2012, and plants were still flowering at the time of seed collection in late 
July.
The majority of unsampled fathers may thus represent relatively early-flowering individuals.
Fortunately, it seems that flowering time is unlikely to depend strongly on location, and as such this is unlikely to bias dispersal estimates.

\subsection{The value of sibship and population data}

We found that information about sibling relationships and information about dispersal improved identification of mating events in simulated data.
In a similar study, \cite{walling2010comparing} compared parentage assignment in a wild deer population inferred from the programs Cervus (which uses parent-offspring trios; \cite{marshall1998statistical}), Colony (which infers parentage and sibling relationships; \cite{wang2009parentage}) and MasterBayes (which infers parentage and population parameters; \cite{hadfield2006towards}).
Both Colony and MasterBayes showed a similar increase in paternity assignment when sibship and population data were included over analyses using the same programs that only used parentage information, but were not able to combine the sources of information.
Here we include both sibship and population information into a single analysis and find that they are useful independent of one another.

There was surprisingly little difference in the accuracy of mating events with increasing family sizes in simulated data (figure \ref{fig:sim-mating}B).
This is counter to expectations by \cite{wang2007parentage} that power should increase with more siblings.
This may be partly due to the structure of the simulations.
We kept the number of mating events fixed, so increasing family size means increasing the total number of progeny.
That in turn means that there are more opportunities for an offspring to be incorrectly assigned to an unrelated father.
In addition, FAPS estimates likelihoods of full sibships by multiplying probabilities of paternity for individuals, which under-utilises the available information about parentage shared between siblings \citep{thompson1987parental}.
As such, it is possible that we have underestimated the value of sibling relationships in this study.

It is helpful to view population data as acting like a Bayesian prior that places different weights on paternity probabilities independent of genotype data (eqn \ref{eqn:probability_model}).
In the case of dispersal, an expectation that dispersal occurs over shorter distances favours candidates close to the mother, which effectively narrows the pool of plausible candidates, and limits the number of unrelated candidates who could incorrectly assigned as a father.
This explains the observation that including dispersal information decreased the number of incorrect mating events and increased the proportion of offspring assigned to missing fathers (figures \ref{fig:sim-mating}B, S8).
It also explains why estimated median dispersal distances were more accurate for narrower dispersal distances (figure \ref{fig:sim-dispersal}).
Our simulations did not investigate the effect of how badly one can misspecify the shape of the dispersal kernel before results are untenable, which would be a useful future avenue.

In this paper we have outlined a general method to combine data about population parameters with analyses of paternity and sibship.
We have modelled dispersal distances this as one such factor affecting mating probabilities to illustrate this, but this clearly underestimates the complexity of interesting mating patterns that could be modelled.
For example, we would expect that a pollen donor's overall floral display size, which may be viewed as a proxy for fecundity, as well as the extent to which flowering phenologies overlap to be positively correlated with the probability of mating with a maternal plant.
Likewise, flower colour may well also play a role in mating, although it is less clear exactly how.
Although it would be feasible to include such data to inform pedigree analyses (see Methods), we have not attempted this here because we currently lack the data to do this in a meaningful way.
Ongoing work currently aims to collect much more detailed demographic and phenotypic information from many more plants, which will allow us to explore these ideas further.

\section{Acknowledgements}

We thank a large number of field volunteers for maintaining the population sampling, and Tom White for assistance with seed collection. We thank Sylvia Rebel for plating tissue for DNA extraction, as well as Sean Stankowski and two anonymous reviewers for feedback on the manuscript.

\section{Data availability}

Data and code to recreate the analyses are available from Zenodo \citep{tom_ellis_2024_10565078}. A supplemental file listing inferred mating events is included as a supplementary file.

\section{Author contributions}

TJE sampled seeds and processed seeds, designed the method, analysed the data and wrote the manuscript.
DLF led the collection of the demographic data of the hybrid zone, designed the SNP panel.
DLF and NHB provided critical feedback on the manuscript. 
%----------------------------------------------------------------------------------------
%	BIBLIOGRAPHY
%----------------------------------------------------------------------------------------

\printbibliography[title={Bibliography}] % Print the bibliography, section title in curly brackets
% \bibliography{tellis.bib}
%----------------------------------------------------------------------------------------

\end{document}
